scripts/getChangelog.py

```diff
@@ -1,90 +1,587 @@
+
+  # Define token
+  print(f"Hello, welcome to the {owner} PR list generator!")
+  time.sleep(0.5)
+  print("First, please enter your GitHub API token.")
+  print("If you don't have one yet, create it at https://github.com/settings/tokens")
+  def get_token():
+    global token
+    data = None
+    while True:
+      if not token:
+        token = getpass.getpass(prompt='Token: ')
+      try:
+        client.inject_token('Bearer ' + token)
+        query = '''
+        query {
+          viewer {
+            login
+          }
+        }
+        '''
+        response = client.execute(query) # This prints the response in case of error, despite any try/except blocks
+        data = json.loads(response)
+        if 'login' in data['data']['viewer']:
+          print(f"Grand, thank you @{data['data']['viewer']['login']}!")
+          print("Do you want learn how to save this token for future use? [y/N]")
+          save = input()
+          if save.lower() == "y":
+            print(textwrap.dedent(f"""
+            To save this token for future use, you need to set it as an environment variable on your system. Here's how you can do it:
+            
+            If you're using bash or zsh, you can add the following line to your shell profile file (.bashrc, .bash_profile or .zshrc):
+            export GITHUB_API_TOKEN='{token}'
+            
+            If you're using Fish shell, you can add the following line to your config.fish file:
+            set -x GITHUB_API_TOKEN '{token}'
+            
+            After adding this line, you'll need to restart your terminal or run 'source ~/.bashrc' (or the appropriate file for your shell) for the changes to take effect.
+            
+            On Windows, you can set environment variables through the System Properties. Here's how:
+            1. Right-click on Computer on the desktop or in the Start menu.
+            2. Choose Properties.
+            3. Click on Advanced system settings.
+            4. Click on Environment Variables.
+            5. Click on New under the User or System variables sections.
+            6. Enter 'GITHUB_API_TOKEN' as the variable name and '{token}' as the variable value.
+            7. Click OK in all windows.
+            
+            Please note that setting environment variables this way will make them available in all future terminal sessions and scripts. Use with caution.
+            """))
+          else:
+            print("Ok, moving on.")
+          break
+      except urllib.error.HTTPError as error_message:
+        if error_message.code == 401:
+          handle_exception(Exception("Invalid GitHub API token, please try again."), data)
+          token = ""
+        else:
+          handle_exception(Exception("Issue executing GraphQL query"), data)
+          token = ""
+      except Exception as error_message:
+        handle_exception(error_message, data)
+        token = ""
+    return token
+
+  if not token:
+    token = get_token()
```

Comment:
Maybe instead `ask_token`? Then it is more clear that it does "UI" and and not calculate stuff